src/main/scala/com/wavesplatform/network/CoordinatorHandler.scala

```diff
@@ -38,88 +38,97 @@ class CoordinatorHandler(checkpointService: CheckpointService,
-      log.error(s"${id(src)} Failure: $failure: $e")
-    case Success(Right(Some(newLocalScore))) =>
-      log.info(s"${id(src)} $success, new local score is $newLocalScore")
-      allChannels.broadcast(LocalScoreChanged(newLocalScore))
-    case Success(Right(None)) => // score has not changed
-    case Success(Left(e)) =>
-      log.warn(s"${id(src)} Left: $failure: $e")
+
+  private def scheduleMiningAndBroadcastScore(score: BigInt): Unit = {
+    miner.scheduleMining()
+    allChannels.broadcast(LocalScoreChanged(score))
   }
 
-  override def channelRead(ctx: ChannelHandlerContext, msg: AnyRef): Unit = {
-    msg match {
-      case c: Checkpoint => broadcastingScore(ctx.channel,
-        "Attempting to process checkpoint",
-        "Successfully processed checkpoint",
-        s"Error processing checkpoint",
-        processCheckpoint(c).map(Some(_)))
-
-      case ExtensionBlocks(blocks) =>
-        blocks.foreach(BlockStats.received(_, BlockStats.Source.Ext, ctx))
-        broadcastingScore(ctx.channel(),
-          s"Attempting to append extension ${formatBlocks(blocks)}",
-          s"Successfully appended extension ${formatBlocks(blocks)}",
-          s"Error appending extension ${formatBlocks(blocks)}",
-          processFork(blocks))
-
-      case b: Block =>
-        BlockStats.received(b, BlockStats.Source.Broadcast, ctx)
-        CoordinatorHandler.blockReceivingLag.safeRecord(System.currentTimeMillis() - b.timestamp)
-        Signed.validateSignatures(b) match {
-          case Left(err) => peerDatabase.blacklistAndClose(ctx.channel(), err.toString)
-          case Right(_) => broadcastingScore(ctx.channel(),
-            s"Attempting to append block ${b.uniqueId}",
-            s"Successfully appended block ${b.uniqueId}",
-            s"Could not append block ${b.uniqueId}", {
-              val blockProcessingResult = processBlock(b, false)
-              blockProcessingResult match {
-                case Left(_) => BlockStats.declined(b, BlockStats.Source.Broadcast)
-                case Right(Some(_)) =>
-                  miner.scheduleMining()
-                  if (b.transactionData.isEmpty)
-                    allChannels.broadcast(BlockForged(b), Some(ctx.channel()))
-                case _ =>
-              }
-              blockProcessingResult
-            }
-          )
-        }
-      case MicroBlockResponse(m) =>
-        val r: Either[Any, Any] = Signed.validateSignatures(m) match {
-          case Right(_) =>
-            Coordinator.processMicroBlock(checkpointService, history, blockchainUpdater, utxStorage)(m)
-              .map { _ =>
-                allChannels.broadcast(MicroBlockInv(m.totalResBlockSig, m.prevResBlockSig), Some(ctx.channel()))
-              }
-          case Left(err) =>
-            peerDatabase.blacklistAndClose(ctx.channel(), err.toString)
-            Left(())
+  private def updateMicroBlockStats(m: MicroBlock): Either[_, Option[Unit]] = processMicroBlock(m) match {
```

Comment:
`updateMicroBlockStats` shouldn't `processMicroBlock`;
is there a way to make it back to `.left.map(..).right.map(..)`?